src/DataGrid.tsx

```diff
@@ -974,6 +974,27 @@ function DataGrid<R, SR, K extends Key>(
     );
   }
 
+  function getRowViewportColumns(rowIdx: number) {
+    const selectedColumn = columns[selectedPosition.idx];
+    if (
+      // idx can be -1 if grouping is enabled
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      selectedColumn !== undefined &&
+      selectedPosition.rowIdx === rowIdx &&
+      !viewportColumns.includes(selectedColumn)
```

Comment:
Might be cheaper to just compare `selectedPosition.idx` with `colOverscanStartIdx`/`colOverscanEndIdx`/`lastFrozenColumnIndex`

Reply:
https://github.com/adazzle/react-data-grid/pull/2562#discussion_r689614421